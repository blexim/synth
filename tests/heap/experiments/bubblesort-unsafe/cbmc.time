file ../../tests/heap/hippo/bubblesort-unsafe.c line 12 function cond1: function `c::nondet' is not declared
**** WARNING: no body for function c::nondet
Command exited with non-zero status 10
2.47user 0.07system 0:02.55elapsed 99%CPU (0avgtext+0avgdata 106280maxresident)k
0inputs+88outputs (0major+82292minor)pagefaults 0swaps
