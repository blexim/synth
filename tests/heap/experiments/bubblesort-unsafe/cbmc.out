CBMC version 4.9 64-bit linux
file ./util.c: Parsing
file ./abstract_transformers.c: Parsing
file ./nested-shakira.c: Parsing
file ../../tests/heap/hippo/bubblesort-unsafe.c: Parsing
Converting
Type-checking abstract_transformers
Type-checking bubblesort-unsafe
Type-checking nested-shakira
Type-checking util
Generating GOTO Program
Adding CPROVER library
Function Pointer Removal
Partial Inlining
Generic Property Instrumentation
Starting Bounded Model Checking
Unwinding loop c::is_minimal.0 iteration 1 file ./abstract_transformers.c line 329 function is_minimal thread 0
Unwinding loop c::is_minimal.0 iteration 2 file ./abstract_transformers.c line 329 function is_minimal thread 0
Unwinding loop c::is_minimal.0 iteration 3 file ./abstract_transformers.c line 329 function is_minimal thread 0
Unwinding loop c::is_minimal.0 iteration 4 file ./abstract_transformers.c line 329 function is_minimal thread 0
Unwinding loop c::is_minimal.0 iteration 5 file ./abstract_transformers.c line 329 function is_minimal thread 0
Unwinding loop c::is_minimal.0 iteration 6 file ./abstract_transformers.c line 329 function is_minimal thread 0
Unwinding loop c::is_minimal.0 iteration 7 file ./abstract_transformers.c line 329 function is_minimal thread 0
Unwinding loop c::is_minimal.0 iteration 8 file ./abstract_transformers.c line 329 function is_minimal thread 0
Unwinding loop c::is_minimal.1 iteration 1 file ./abstract_transformers.c line 337 function is_minimal thread 0
Unwinding loop c::is_minimal.1 iteration 2 file ./abstract_transformers.c line 337 function is_minimal thread 0
Unwinding loop c::is_minimal.1 iteration 3 file ./abstract_transformers.c line 337 function is_minimal thread 0
Unwinding loop c::is_minimal.1 iteration 4 file ./abstract_transformers.c line 337 function is_minimal thread 0
Unwinding loop c::is_minimal.1 iteration 5 file ./abstract_transformers.c line 337 function is_minimal thread 0
Unwinding loop c::is_minimal.1 iteration 6 file ./abstract_transformers.c line 337 function is_minimal thread 0
Unwinding loop c::is_minimal.1 iteration 7 file ./abstract_transformers.c line 337 function is_minimal thread 0
Unwinding loop c::is_minimal.1 iteration 8 file ./abstract_transformers.c line 337 function is_minimal thread 0
Unwinding loop c::is_minimal.2 iteration 1 file ./abstract_transformers.c line 338 function is_minimal thread 0
Unwinding loop c::is_minimal.2 iteration 2 file ./abstract_transformers.c line 338 function is_minimal thread 0
Unwinding loop c::is_minimal.2 iteration 3 file ./abstract_transformers.c line 338 function is_minimal thread 0
Unwinding loop c::is_minimal.2 iteration 4 file ./abstract_transformers.c line 338 function is_minimal thread 0
Unwinding loop c::is_minimal.2 iteration 5 file ./abstract_transformers.c line 338 function is_minimal thread 0
Unwinding loop c::is_minimal.2 iteration 6 file ./abstract_transformers.c line 338 function is_minimal thread 0
Unwinding loop c::is_minimal.2 iteration 7 file ./abstract_transformers.c line 338 function is_minimal thread 0
Unwinding loop c::is_minimal.2 iteration 8 file ./abstract_transformers.c line 338 function is_minimal thread 0
Unwinding loop c::is_minimal.3 iteration 1 file ./abstract_transformers.c line 349 function is_minimal thread 0
Unwinding loop c::is_minimal.3 iteration 2 file ./abstract_transformers.c line 349 function is_minimal thread 0
Unwinding loop c::is_minimal.3 iteration 3 file ./abstract_transformers.c line 349 function is_minimal thread 0
Unwinding loop c::is_minimal.3 iteration 4 file ./abstract_transformers.c line 349 function is_minimal thread 0
Unwinding loop c::is_minimal.3 iteration 5 file ./abstract_transformers.c line 349 function is_minimal thread 0
Unwinding loop c::is_minimal.3 iteration 6 file ./abstract_transformers.c line 349 function is_minimal thread 0
Unwinding loop c::is_minimal.3 iteration 7 file ./abstract_transformers.c line 349 function is_minimal thread 0
Unwinding loop c::is_minimal.4 iteration 1 file ./abstract_transformers.c line 345 function is_minimal thread 0
Unwinding loop c::is_minimal.3 iteration 1 file ./abstract_transformers.c line 349 function is_minimal thread 0
Unwinding loop c::is_minimal.3 iteration 2 file ./abstract_transformers.c line 349 function is_minimal thread 0
Unwinding loop c::is_minimal.3 iteration 3 file ./abstract_transformers.c line 349 function is_minimal thread 0
Unwinding loop c::is_minimal.3 iteration 4 file ./abstract_transformers.c line 349 function is_minimal thread 0
Unwinding loop c::is_minimal.3 iteration 5 file ./abstract_transformers.c line 349 function is_minimal thread 0
Unwinding loop c::is_minimal.3 iteration 6 file ./abstract_transformers.c line 349 function is_minimal thread 0
Unwinding loop c::is_minimal.3 iteration 7 file ./abstract_transformers.c line 349 function is_minimal thread 0
Unwinding loop c::is_minimal.4 iteration 2 file ./abstract_transformers.c line 345 function is_minimal thread 0
Unwinding loop c::is_minimal.3 iteration 1 file ./abstract_transformers.c line 349 function is_minimal thread 0
Unwinding loop c::is_minimal.3 iteration 2 file ./abstract_transformers.c line 349 function is_minimal thread 0
Unwinding loop c::is_minimal.3 iteration 3 file ./abstract_transformers.c line 349 function is_minimal thread 0
Unwinding loop c::is_minimal.3 iteration 4 file ./abstract_transformers.c line 349 function is_minimal thread 0
Unwinding loop c::is_minimal.3 iteration 5 file ./abstract_transformers.c line 349 function is_minimal thread 0
Unwinding loop c::is_minimal.3 iteration 6 file ./abstract_transformers.c line 349 function is_minimal thread 0
Unwinding loop c::is_minimal.3 iteration 7 file ./abstract_transformers.c line 349 function is_minimal thread 0
Unwinding loop c::is_minimal.4 iteration 3 file ./abstract_transformers.c line 345 function is_minimal thread 0
Unwinding loop c::is_minimal.3 iteration 1 file ./abstract_transformers.c line 349 function is_minimal thread 0
Unwinding loop c::is_minimal.3 iteration 2 file ./abstract_transformers.c line 349 function is_minimal thread 0
Unwinding loop c::is_minimal.3 iteration 3 file ./abstract_transformers.c line 349 function is_minimal thread 0
Unwinding loop c::is_minimal.3 iteration 4 file ./abstract_transformers.c line 349 function is_minimal thread 0
Unwinding loop c::is_minimal.3 iteration 5 file ./abstract_transformers.c line 349 function is_minimal thread 0
Unwinding loop c::is_minimal.3 iteration 6 file ./abstract_transformers.c line 349 function is_minimal thread 0
Unwinding loop c::is_minimal.3 iteration 7 file ./abstract_transformers.c line 349 function is_minimal thread 0
Unwinding loop c::is_minimal.4 iteration 4 file ./abstract_transformers.c line 345 function is_minimal thread 0
Unwinding loop c::is_minimal.5 iteration 1 file ./abstract_transformers.c line 382 function is_minimal thread 0
Unwinding loop c::is_minimal.5 iteration 2 file ./abstract_transformers.c line 382 function is_minimal thread 0
Unwinding loop c::is_minimal.5 iteration 3 file ./abstract_transformers.c line 382 function is_minimal thread 0
Unwinding loop c::is_minimal.5 iteration 4 file ./abstract_transformers.c line 382 function is_minimal thread 0
Unwinding loop c::is_minimal.5 iteration 5 file ./abstract_transformers.c line 382 function is_minimal thread 0
Unwinding loop c::is_minimal.5 iteration 6 file ./abstract_transformers.c line 382 function is_minimal thread 0
Unwinding loop c::is_minimal.5 iteration 7 file ./abstract_transformers.c line 382 function is_minimal thread 0
Unwinding loop c::is_minimal.5 iteration 8 file ./abstract_transformers.c line 382 function is_minimal thread 0
Unwinding loop c::path_len.0 iteration 1 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 2 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 3 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 4 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 5 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 6 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 7 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 8 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 9 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 1 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 2 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 3 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 4 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 5 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 6 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 7 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 8 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 9 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 1 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 2 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 3 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 4 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 5 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 6 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 7 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 8 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 9 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 1 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 2 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 3 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 4 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 5 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 6 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 7 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 8 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 9 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 1 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 2 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 3 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 4 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 5 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 6 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 7 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 8 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 9 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 1 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 2 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 3 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 4 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 5 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 6 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 7 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 8 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 9 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 1 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 2 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 3 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 4 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 5 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 6 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 7 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 8 file ./abstract_transformers.c line 408 function path_len thread 0
Unwinding loop c::path_len.0 iteration 9 file ./abstract_transformers.c line 408 function path_len thread 0
size of program expression: 7088 steps
slicing removed 3207 assignments
Generated 7 VCC(s), 7 remaining after simplification
Passing problem to propositional reduction
Running propositional reduction
Post-processing
Solving with Glucose 2.2 with simplifier
42693 variables, 159439 clauses
c |  Eliminated clauses:           1.59 Mb                                                                |
c last restart ## conflicts  :  493 166 
SAT checker: negated claim is SATISFIABLE, i.e., does not hold
Runtime decision procedure: 0.606s
Building error trace

Counterexample:

State 6 file ./nested-shakira.c line 17 function main thread 0
----------------------------------------------------
  h={ .succ={ 0, 0, 5, 2, 3, 1, 7, 7 }, .dist={ 0, 1, 31, 1, 27, 30, 1, 31 }, .ptr={ 0, 1, 1, 1 },
    .nnodes=2 } ({ { 00000, 00000, 00101, 00010, 00011, 00001, 00111, 00111 }, { 00000, 00001, 11111, 00001, 11011, 11110, 00001, 11111 }, { 00000, 00001, 00001, 00001 }, 00010 })

State 7 file ./nested-shakira.c line 17 function main thread 0
----------------------------------------------------
  t3={ .succ={ 3, 2, 5, 7, 3, 0, 1, 19 }, .dist={ 0, 31, 31, 0, 31, 0, 1, 12 }, .ptr={ 0, 2, 1, 0 },
    .nnodes=0 } ({ { 00011, 00010, 00101, 00111, 00011, 00000, 00001, 10011 }, { 00000, 11111, 11111, 00000, 11111, 00000, 00001, 01100 }, { 00000, 00010, 00001, 00000 }, 00000 })

State 13 file ./abstract_transformers.c line 17 function deref thread 0
----------------------------------------------------
  return_value_deref$1=0 (00000)

State 19 file ./abstract_transformers.c line 27 function next thread 0
----------------------------------------------------
  return_value_next$2=0 (00000)

State 25 file ./abstract_transformers.c line 36 function dist thread 0
----------------------------------------------------
  return_value_dist$3=0 (00000)

State 30 file ./abstract_transformers.c line 328 function is_minimal thread 0
----------------------------------------------------
  is_named={ 0, 0, 0, 0, 0, 0, 0, 0 } ({ 00000, 00000, 00000, 00000, 00000, 00000, 00000, 00000 })

State 32 file ./abstract_transformers.c line 329 function is_minimal thread 0
----------------------------------------------------
  is_named={ 0, 0, 0, 0, 0, 0, 0, 0 } ({ 00000, 00000, 00000, 00000, 00000, 00000, 00000, 00000 })

State 34 file ./abstract_transformers.c line 329 function is_minimal thread 0
----------------------------------------------------
  is_named={ 0, 0, 0, 0, 0, 0, 0, 0 } ({ 00000, 00000, 00000, 00000, 00000, 00000, 00000, 00000 })

State 36 file ./abstract_transformers.c line 329 function is_minimal thread 0
----------------------------------------------------
  is_named={ 0, 0, 0, 0, 0, 0, 0, 0 } ({ 00000, 00000, 00000, 00000, 00000, 00000, 00000, 00000 })

State 38 file ./abstract_transformers.c line 329 function is_minimal thread 0
----------------------------------------------------
  is_named={ 0, 0, 0, 0, 0, 0, 0, 0 } ({ 00000, 00000, 00000, 00000, 00000, 00000, 00000, 00000 })

State 40 file ./abstract_transformers.c line 329 function is_minimal thread 0
----------------------------------------------------
  is_named={ 0, 0, 0, 0, 0, 0, 0, 0 } ({ 00000, 00000, 00000, 00000, 00000, 00000, 00000, 00000 })

State 42 file ./abstract_transformers.c line 329 function is_minimal thread 0
----------------------------------------------------
  is_named={ 0, 0, 0, 0, 0, 0, 0, 0 } ({ 00000, 00000, 00000, 00000, 00000, 00000, 00000, 00000 })

State 44 file ./abstract_transformers.c line 329 function is_minimal thread 0
----------------------------------------------------
  is_named={ 0, 0, 0, 0, 0, 0, 0, 0 } ({ 00000, 00000, 00000, 00000, 00000, 00000, 00000, 00000 })

State 46 file ./abstract_transformers.c line 329 function is_minimal thread 0
----------------------------------------------------
  is_named={ 0, 0, 0, 0, 0, 0, 0, 0 } ({ 00000, 00000, 00000, 00000, 00000, 00000, 00000, 00000 })

State 51 file ./abstract_transformers.c line 333 function is_minimal thread 0
----------------------------------------------------
  is_reachable={ 0, 0, 0, 0, 0, 0, 0, 0 } ({ 00000, 00000, 00000, 00000, 00000, 00000, 00000, 00000 })

State 52 file ./abstract_transformers.c line 334 function is_minimal thread 0
----------------------------------------------------
  indegree={ 0, 0, 0, 0, 0, 0, 0, 0 } ({ 00000, 00000, 00000, 00000, 00000, 00000, 00000, 00000 })

State 54 file ./abstract_transformers.c line 337 function is_minimal thread 0
----------------------------------------------------
  is_reachable={ 0, 0, 0, 0, 0, 0, 0, 0 } ({ 00000, 00000, 00000, 00000, 00000, 00000, 00000, 00000 })

State 56 file ./abstract_transformers.c line 337 function is_minimal thread 0
----------------------------------------------------
  is_reachable={ 0, 0, 0, 0, 0, 0, 0, 0 } ({ 00000, 00000, 00000, 00000, 00000, 00000, 00000, 00000 })

State 58 file ./abstract_transformers.c line 337 function is_minimal thread 0
----------------------------------------------------
  is_reachable={ 0, 0, 0, 0, 0, 0, 0, 0 } ({ 00000, 00000, 00000, 00000, 00000, 00000, 00000, 00000 })

State 60 file ./abstract_transformers.c line 337 function is_minimal thread 0
----------------------------------------------------
  is_reachable={ 0, 0, 0, 0, 0, 0, 0, 0 } ({ 00000, 00000, 00000, 00000, 00000, 00000, 00000, 00000 })

State 62 file ./abstract_transformers.c line 337 function is_minimal thread 0
----------------------------------------------------
  is_reachable={ 0, 0, 0, 0, 0, 0, 0, 0 } ({ 00000, 00000, 00000, 00000, 00000, 00000, 00000, 00000 })

State 64 file ./abstract_transformers.c line 337 function is_minimal thread 0
----------------------------------------------------
  is_reachable={ 0, 0, 0, 0, 0, 0, 0, 0 } ({ 00000, 00000, 00000, 00000, 00000, 00000, 00000, 00000 })

State 66 file ./abstract_transformers.c line 337 function is_minimal thread 0
----------------------------------------------------
  is_reachable={ 0, 0, 0, 0, 0, 0, 0, 0 } ({ 00000, 00000, 00000, 00000, 00000, 00000, 00000, 00000 })

State 68 file ./abstract_transformers.c line 337 function is_minimal thread 0
----------------------------------------------------
  is_reachable={ 0, 0, 0, 0, 0, 0, 0, 0 } ({ 00000, 00000, 00000, 00000, 00000, 00000, 00000, 00000 })

State 74 file ./abstract_transformers.c line 338 function is_minimal thread 0
----------------------------------------------------
  indegree={ 0, 0, 0, 0, 0, 0, 0, 0 } ({ 00000, 00000, 00000, 00000, 00000, 00000, 00000, 00000 })

State 76 file ./abstract_transformers.c line 338 function is_minimal thread 0
----------------------------------------------------
  indegree={ 0, 0, 0, 0, 0, 0, 0, 0 } ({ 00000, 00000, 00000, 00000, 00000, 00000, 00000, 00000 })

State 78 file ./abstract_transformers.c line 338 function is_minimal thread 0
----------------------------------------------------
  indegree={ 0, 0, 0, 0, 0, 0, 0, 0 } ({ 00000, 00000, 00000, 00000, 00000, 00000, 00000, 00000 })

State 80 file ./abstract_transformers.c line 338 function is_minimal thread 0
----------------------------------------------------
  indegree={ 0, 0, 0, 0, 0, 0, 0, 0 } ({ 00000, 00000, 00000, 00000, 00000, 00000, 00000, 00000 })

State 82 file ./abstract_transformers.c line 338 function is_minimal thread 0
----------------------------------------------------
  indegree={ 0, 0, 0, 0, 0, 0, 0, 0 } ({ 00000, 00000, 00000, 00000, 00000, 00000, 00000, 00000 })

State 84 file ./abstract_transformers.c line 338 function is_minimal thread 0
----------------------------------------------------
  indegree={ 0, 0, 0, 0, 0, 0, 0, 0 } ({ 00000, 00000, 00000, 00000, 00000, 00000, 00000, 00000 })

State 86 file ./abstract_transformers.c line 338 function is_minimal thread 0
----------------------------------------------------
  indegree={ 0, 0, 0, 0, 0, 0, 0, 0 } ({ 00000, 00000, 00000, 00000, 00000, 00000, 00000, 00000 })

State 88 file ./abstract_transformers.c line 338 function is_minimal thread 0
----------------------------------------------------
  indegree={ 0, 0, 0, 0, 0, 0, 0, 0 } ({ 00000, 00000, 00000, 00000, 00000, 00000, 00000, 00000 })

State 96 file ./abstract_transformers.c line 17 function deref thread 0
----------------------------------------------------
  n=0 (00000)

State 98 file ./abstract_transformers.c line 347 function is_minimal thread 0
----------------------------------------------------
  is_named[0]=1 (00001)

State 103 file ./abstract_transformers.c line 355 function is_minimal thread 0
----------------------------------------------------
  n=0 (00000)

State 106 file ./abstract_transformers.c line 36 function dist thread 0
----------------------------------------------------
  return_value_dist$1=0 (00000)

State 111 file ./abstract_transformers.c line 368 function is_minimal thread 0
----------------------------------------------------
  last_reachable=0 (00000)

State 112 file ./abstract_transformers.c line 370 function is_minimal thread 0
----------------------------------------------------
  is_reachable[0]=1 (00001)

State 115 file ./abstract_transformers.c line 373 function is_minimal thread 0
----------------------------------------------------
  n=0 (00000)

State 118 file ./abstract_transformers.c line 27 function next thread 0
----------------------------------------------------
  n=0 (00000)

State 120 file ./abstract_transformers.c line 376 function is_minimal thread 0
----------------------------------------------------
  indegree[0]=1 (00001)

State 139 file ./abstract_transformers.c line 17 function deref thread 0
----------------------------------------------------
  n=1 (00001)

State 141 file ./abstract_transformers.c line 347 function is_minimal thread 0
----------------------------------------------------
  is_named[1]=1 (00001)

State 146 file ./abstract_transformers.c line 355 function is_minimal thread 0
----------------------------------------------------
  n=1 (00001)

State 149 file ./abstract_transformers.c line 36 function dist thread 0
----------------------------------------------------
  return_value_dist$1=1 (00001)

State 155 file ./abstract_transformers.c line 359 function is_minimal thread 0
----------------------------------------------------
  n=1 (00001)

State 158 file ./abstract_transformers.c line 36 function dist thread 0
----------------------------------------------------
  return_value_dist$2=1 (00001)

State 162 file ./abstract_transformers.c line 368 function is_minimal thread 0
----------------------------------------------------
  last_reachable=1 (00001)

State 163 file ./abstract_transformers.c line 370 function is_minimal thread 0
----------------------------------------------------
  is_reachable[1]=1 (00001)

State 164 file ./abstract_transformers.c line 371 function is_minimal thread 0
----------------------------------------------------
  nreachable=2 (00010)

State 167 file ./abstract_transformers.c line 373 function is_minimal thread 0
----------------------------------------------------
  n=1 (00001)

State 170 file ./abstract_transformers.c line 27 function next thread 0
----------------------------------------------------
  n=0 (00000)

State 172 file ./abstract_transformers.c line 376 function is_minimal thread 0
----------------------------------------------------
  indegree[0]=2 (00010)

State 191 file ./abstract_transformers.c line 17 function deref thread 0
----------------------------------------------------
  n=1 (00001)

State 193 file ./abstract_transformers.c line 347 function is_minimal thread 0
----------------------------------------------------
  is_named[1]=1 (00001)

State 213 file ./abstract_transformers.c line 17 function deref thread 0
----------------------------------------------------
  n=1 (00001)

State 215 file ./abstract_transformers.c line 347 function is_minimal thread 0
----------------------------------------------------
  is_named[1]=1 (00001)

State 261 file ./abstract_transformers.c line 324 function valid_abstract_heap thread 0
----------------------------------------------------
  return_value_valid_abstract_heap$1=1 (00000000000000000000000000000001)

State 270 file ./abstract_transformers.c line 7 function copy_abstract thread 0
----------------------------------------------------
  t1={ .succ={ 0, 0, 5, 2, 3, 1, 7, 7 }, .dist={ 0, 1, 31, 1, 27, 30, 1, 31 }, .ptr={ 0, 1, 1, 1 },
    .nnodes=2 } ({ { 00000, 00000, 00101, 00010, 00011, 00001, 00111, 00111 }, { 00000, 00001, 11111, 00001, 11011, 11110, 00001, 11111 }, { 00000, 00001, 00001, 00001 }, 00010 })

State 275 file ./abstract_transformers.c line 17 function deref thread 0
----------------------------------------------------
  py=1 (00001)

State 279 file ./abstract_transformers.c line 83 function abstract_assign thread 0
----------------------------------------------------
  n=1 (00001)

State 281 file ./abstract_transformers.c line 50 function destructive_assign_ptr thread 0
----------------------------------------------------
  t1.ptr[1]=1 (00001)

State 285 file ../../tests/heap/hippo/bubblesort-unsafe.c line 9 function pre1 thread 0
----------------------------------------------------
  return_value_pre1$3=0 (00000000000000000000000000000000)

State 295 file ./abstract_transformers.c line 17 function deref thread 0
----------------------------------------------------
  n=1 (00001)

State 300 file ./abstract_transformers.c line 17 function deref thread 0
----------------------------------------------------
  yn=1 (00001)

State 306 file ../../tests/heap/hippo/bubblesort-unsafe.c line 53 function inv1 thread 0
----------------------------------------------------
  return_value_inv1$8=1 (00000000000000000000000000000001)

State 314 file ../../tests/heap/hippo/bubblesort-unsafe.c line 12 function cond1 thread 0
----------------------------------------------------
  return_value_nondet$1=0 (00000000000000000000000000000000)

State 316 file ../../tests/heap/hippo/bubblesort-unsafe.c line 12 function cond1 thread 0
----------------------------------------------------
  return_value_cond1$7=0 (00000000000000000000000000000000)

State 326 file ./abstract_transformers.c line 17 function deref thread 0
----------------------------------------------------
  n=1 (00001)

State 331 file ./abstract_transformers.c line 17 function deref thread 0
----------------------------------------------------
  yn=1 (00001)

State 337 file ../../tests/heap/hippo/bubblesort-unsafe.c line 49 function assertion thread 0
----------------------------------------------------
  return_value_assertion$6=1 (00000000000000000000000000000001)

State 348 file ./abstract_transformers.c line 17 function deref thread 0
----------------------------------------------------
  n=1 (00001)

State 353 file ./abstract_transformers.c line 17 function deref thread 0
----------------------------------------------------
  yn=1 (00001)

State 359 file ../../tests/heap/hippo/bubblesort-unsafe.c line 57 function inv2 thread 0
----------------------------------------------------
  return_value_inv2$13=1 (00000000000000000000000000000001)

State 369 file ./abstract_transformers.c line 17 function deref thread 0
----------------------------------------------------
  return_value_deref$1=1 (00001)

State 372 file ./abstract_transformers.c line 431 function is_null thread 0
----------------------------------------------------
  return_value_is_null$1=0 (00000)

State 375 file ../../tests/heap/hippo/bubblesort-unsafe.c line 20 function cond2 thread 0
----------------------------------------------------
  return_value_cond2$12=1 (00000000000000000000000000000001)

State 385 file ./abstract_transformers.c line 17 function deref thread 0
----------------------------------------------------
  return_value_deref$1=1 (00001)

State 388 file ./abstract_transformers.c line 431 function is_null thread 0
----------------------------------------------------
  return_value_is_null$1=0 (00000)

State 396 file ./abstract_transformers.c line 17 function deref thread 0
----------------------------------------------------
  py=1 (00001)

State 400 file ./abstract_transformers.c line 124 function abstract_lookup thread 0
----------------------------------------------------
  n=1 (00001)

State 403 file ./abstract_transformers.c line 27 function next thread 0
----------------------------------------------------
  yn=0 (00000)

State 408 file ./abstract_transformers.c line 128 function abstract_lookup thread 0
----------------------------------------------------
  n=1 (00001)

State 411 file ./abstract_transformers.c line 36 function dist thread 0
----------------------------------------------------
  y_yn_dist=1 (00001)

State 415 file ./abstract_transformers.c line 7 function copy_abstract thread 0
----------------------------------------------------
  t1={ .succ={ 0, 0, 5, 2, 3, 1, 7, 7 }, .dist={ 0, 1, 31, 1, 27, 30, 1, 31 }, .ptr={ 0, 1, 1, 1 },
    .nnodes=2 } ({ { 00000, 00000, 00101, 00010, 00011, 00001, 00111, 00111 }, { 00000, 00001, 11111, 00001, 11011, 11110, 00001, 11111 }, { 00000, 00001, 00001, 00001 }, 00010 })

State 420 file ./abstract_transformers.c line 145 function abstract_lookup thread 0
----------------------------------------------------
  n=0 (00000)

State 422 file ./abstract_transformers.c line 50 function destructive_assign_ptr thread 0
----------------------------------------------------
  t1.ptr[3]=0 (00000)

State 431 file ./abstract_transformers.c line 17 function deref thread 0
----------------------------------------------------
  return_value_deref$1=1 (00001)

State 434 file ./abstract_transformers.c line 431 function is_null thread 0
----------------------------------------------------
  return_value_is_null$2=0 (00000)

State 442 file ./abstract_transformers.c line 17 function deref thread 0
----------------------------------------------------
  return_value_deref$1=0 (00000)

State 445 file ./abstract_transformers.c line 431 function is_null thread 0
----------------------------------------------------
  return_value_is_null$3=1 (00001)

State 447 file ../../tests/heap/hippo/bubblesort-unsafe.c line 32 function body2 thread 0
----------------------------------------------------
  tmp_if_expr$4=TRUE (1)

Violated property:
  file ./nested-shakira.c line 45 function main
  assertion body2(&h, &t3)
  FALSE

VERIFICATION FAILED
